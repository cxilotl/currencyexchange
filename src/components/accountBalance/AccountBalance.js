import React from "react";
import PropTypes from 'prop-types';
import cssStyles from './AccountBalance.module.scss';

const AccountBalance = ({ currencySymbol, balance}) => {
  return(
    <div className={ cssStyles.layout }>
      <p className={ cssStyles.text }>You have <span data-testid="account-balance">{ currencySymbol }{ balance.toFixed(2) }</span></p>
    </div>
  );
};

AccountBalance.propTypes = {
  currencySymbol: PropTypes.string,
  balance: PropTypes.number
};

export default AccountBalance;