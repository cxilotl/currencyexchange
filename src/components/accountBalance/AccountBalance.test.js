import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import AccountBalance from './AccountBalance';

describe('AccountBalance', () => {
  const currencySymbol = '£';
  const balance = 53.5;

  it('Snapshot', () => {
    const component = renderer.create(
      <AccountBalance
        currencySymbol={ currencySymbol }
        balance={ balance }
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Given a AccountBalance element, it', () => {
    afterEach(cleanup);

    it('should render a balance with 2 decimals', () => {
      const { getByTestId } = render(
        <AccountBalance
          currencySymbol={ currencySymbol }
          balance={ balance }
        />
      );
      const accountBalance = getByTestId('account-balance');
      expect(accountBalance.innerHTML).toEqual('£53.50');
    });
  });
});