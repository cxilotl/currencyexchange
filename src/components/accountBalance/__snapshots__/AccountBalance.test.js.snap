// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountBalance Snapshot 1`] = `
<div
  className="layout"
>
  <p
    className="text"
  >
    You have 
    <span
      data-testid="account-balance"
    >
      Â£
      53.50
    </span>
  </p>
</div>
`;
